tools/integration-tests/tester/framework/config/config.go

```diff
@@ -31,76 +49,181 @@ type GoShimmer struct {
 	DRNG
 }
 
+// NewGoShimmer creates a GoShimmer config initialized with default values.
+func NewGoShimmer() (config GoShimmer) {
+	config = GoShimmer{}
+	fillStructFromDefaultTag(&config)
+	return
+}
+
+// fillStructFromDefaultTag recursively explores the given struct pointer and sets values of fields to the `default` as
+// specified in the struct's tags.
+func fillStructFromDefaultTag(s interface{}) {
+	val := reflect.ValueOf(s).Elem()
```

Comment:
https://goshimmer.docs.iota.org/implementation_design/configuration_parameters.html should include a discussion about `reflect`

Reply:
What exactly do you mean?

Reply:
The documentation should be changed to reflect that parameters should be defined in a type `ParametersDefinition` and `Parameters` now becomes a var of that type. But documenting on how the internals work with `reflect` I think will go in too much detail imho.